src/lib/db/role-store.ts

```diff
@@ -141,6 +148,15 @@ export default class RoleStore implements IRoleStore {
             .orWhere('type', 'project');
     }
 
+    async getProjectRolesCount(): Promise<number> {
+        return this.db
+            .from<IRole>(T.ROLES)
+            .count('*')
+            .where('type', 'custom')
+            .orWhere('type', 'project')
+            .then((res) => Number(res[0].count));
```

Comment:
I'm curious about this. I guess since it's count it always returns a row? But not every knex method does this, as far as I know. Sometimes this causes problems because people try to copy what they have seen elsewhere, and expect it to work the same. Not sure if we should do something about it, but I'm raising the point.

Reply:
In knex a `count()` will return a result of type: `Record<string, number | string>`. So we know it will be a single row with a "key" named "count" and the value to be a `number` or a stringed number (`42` vs `"42"`).